python/ray/data/block.py

```diff
@@ -370,7 +370,18 @@ def batch_to_block(batch: DataBatch) -> Block:
 
                 # TODO(ekl) once we support Python objects within Arrow blocks, we
                 # don't need this fallback path.
-                return pd.DataFrame(dict(batch))
+                if not isinstance(batch, dict):
+                    return pd.DataFrame(dict(batch))
+
+                # batch is type of dict[str, np.ndarray]
+                # check for arrays with ndim > 1
+                if all(np.ndim(val) == 1 for val in batch.values()):
+                    return pd.DataFrame(batch)
```

Comment:
can you clarify why we need to add the check for columns with more than 1 dimension?

Reply:
`pandas` does not allow construction of DataFrames from dictionaries with column arrays with dimensions greater than 1:
https://github.com/pandas-dev/pandas/blob/61d2056e4251b8d1c387f2577f0407062b9ab903/pandas/core/internals/construction.py#L663-L664

Stacktrace:
```
  File "/Users/michaelhly/Projects/ray/python/ray/data/block.py", line 378, in batch_to_block
    return pd.DataFrame(batch)
  File "/Users/michaelhly/miniforge3/envs/ray/lib/python3.9/site-packages/pandas/core/frame.py", line 736, in __init__
    mgr = dict_to_mgr(data, index, columns, dtype=dtype, copy=copy, typ=manager)
  File "/Users/michaelhly/miniforge3/envs/ray/lib/python3.9/site-packages/pandas/core/internals/construction.py", line 503, in dict_to_mgr
    return arrays_to_mgr(arrays, columns, index, dtype=dtype, typ=typ, consolidate=copy)
  File "/Users/michaelhly/miniforge3/envs/ray/lib/python3.9/site-packages/pandas/core/internals/construction.py", line 114, in arrays_to_mgr
    index = _extract_index(arrays)
  File "/Users/michaelhly/miniforge3/envs/ray/lib/python3.9/site-packages/pandas/core/internals/construction.py", line 664, in _extract_index
    raise ValueError("Per-column arrays must each be 1-dimensional")
ValueError: Per-column arrays must each be 1-dimensional
```

Reply:
I can also revert these changes, but I thought it would be nice to know which column failed when converting batch -> block